ports/liborigin/portfile.cmake

```
@@ -0,0 +1,35 @@
+vcpkg_from_sourceforge(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO liborigin
+    REF liborigin/3.0
+    FILENAME liborigin-${VERSION}.tar.gz
+    SHA512 9fb5ae6d8aa8fb54e659482f8f5dc581b8d0ace2ebca7bb9f092b7ec753049d497491eb47ad89b12c8ddf7e19dc47f76e76c51ace789366370bd056d99e091ee
+)
+
+if(VCPKG_CRT_LINKAGE STREQUAL "dynamic")
+  set(LIB_OPTION "-DBUILD_SHARED_LIBS=ON -DBUILD_STATIC_LIBS=OFF")
+else()
+  set(LIB_OPTION "-DBUILD_SHARED_LIBS=OFF -DBUILD_STATIC_LIBS=ON")
```

```suggestion
  set(LIB_OPTION "-DBUILD_STATIC_LIBS=OFF")
else()
  set(LIB_OPTION "-DBUILD_STATIC_LIBS=ON")
```
`vcpkg_cmake_configure` needs a (CMake) list of options, and it already handles `BUILD_SHARED_LIBS`.

---

This project disable static libs with mavc when shared libs are enabled (which is on by default)

---

So IIUC now this block isn't needed at all because `vcpkg_cmake_configure` controls `BUILD_SHARED_LIBS`.

---

Both static and shared libs are enabled by default.

---

Then we are back to my suggested change... Where is the problem?

---

Both are enabled by default except with MSVC where static libs are disabled when shared libs are enabled.
Why does someone need a static library when he wants only shared ones and vice versa.

---

> > Why does someone need a static library when he wants only shared ones and vice versa.
> 
> This doesn't need be discussed. I didn't suggest to remove `BUILD_STATIC_LIBS`. I pointed out that
> - `BUILD_SHARED_LIBS` is already handled and doesn't need to be added one more time,

Does vcpkg pass BUILD_SHARED_LIBS=OFF with static?
> - these variable must reflect `VCPKG_LIBRARY_LINKAGE`, not `VCPKG_CRT_LINKAGE`,

OK, I am far from my PC, will change when come home.
> - `OPTIONS` can't be command line strings ( `"-DA=1 -DB=2"`) but must be CMake lists (`"-DA=1" "-DB=2"`).

OK, will fix it.

---

> Does vcpkg pass BUILD_SHARED_LIBS=OFF with static?

Yes.
https://github.com/microsoft/vcpkg/blob/90542a476c8ef3288039e2443a775e48db279173/ports/vcpkg-cmake/vcpkg_cmake_configure.cmake#L139-L147